map/api_mark_point.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include "map/user_mark.hpp"
+#include "map/styled_point.hpp"
+
+class ApiMarkPoint : public StyledPoint
```

Comment:
What is the reason to have 2 separate classes for StyledPoint and ApiMarkPoint?
I see it as "extending" existing ApiMarkPoint implementation (we don't need old one, only the basic icon if style was not specified).